zenodo/modules/spam/utils.py

```diff
@@ -75,20 +82,22 @@ def send_spam_admin_email(user, deposit=None, community=None):
 def check_and_handle_spam(community=None, deposit=None, retry=True):
     """Checks community/deposit metadata for spam."""
     try:
-        if current_app.config.get('ZENODO_SPAM_MODEL_LOCATION'):
-            if community:
-                task = check_metadata_for_spam.delay(
-                    community_id=community.id)
-                user_id = community.id_user
-            if deposit:
-                task = check_metadata_for_spam.delay(dep_id=str(deposit.id))
-                user_id = deposit['owners'][0]
-            spam_proba = task.get(timeout=current_app.config[
-                'ZENODO_SPAM_CHECK_TIMEOUT'])
-        else:
-            spam_proba = 0
-        if spam_proba > current_app.config['ZENODO_SPAM_THRESHOLD']:
-            if not Permission(ActionNeed('admin-access')).can():
+        if not is_safelisted_user(current_user) or \
+            not Permission(ActionNeed('admin-access')).can():
```

Comment:
```suggestion
        is_admin = Permission(ActionNeed('admin-access')).can()
        if not is_safelisted_user(current_user) or \
            not is_admin:
```

And further for readability